src/controller/java/OTAProviderDelegateBridge.cpp

```diff
@@ -0,0 +1,375 @@
+    Optional<bool> requestorCanConsent                               = commandData.requestorCanConsent;
+    Optional<chip::ByteSpan> metadataForProvider                     = commandData.metadataForProvider;
+
+    bool isBDXProtocolSupported = false;
+
+    auto iterator = commandData.protocolsSupported.begin();
+    while (iterator.Next())
+    {
+        OTADownloadProtocol protocol = iterator.GetValue();
+        if (protocol == OTADownloadProtocol::kBDXSynchronous)
+        {
+            isBDXProtocolSupported = true;
+            break;
+        }
+    }
+
+    ConcreteCommandPath cachedCommandPath(commandPath.mEndpointId, commandPath.mClusterId, commandPath.mCommandId);
+
+    // The logic we are following here is if none of the protocols supported by the requestor are supported by us, we
+    // can't transfer the image even if we had an image available and we would return a Protocol Not Supported status.
+    // Assumption here is the requestor would send us a list of all the protocols it supports. If one/more of the
+    // protocols supported by the requestor are supported by us, we check if an image is not available due to various
+    // reasons - image not available, delegate reporting busy, we will respond with the status in the delegate response.
+    // If update is available, we try to prepare for transfer and build the uri in the response with a status of Image
+    // Available
+
+    // If the protocol requested is not supported, return status - Protocol Not Supported
+    if (!isBDXProtocolSupported)
+    {
+        Commands::QueryImageResponse::Type response;
+        response.status = OTAQueryStatus::kDownloadProtocolNotSupported;
+        commandObj->AddResponse(cachedCommandPath, response);
+        sendOTAQueryFailure(static_cast<uint8_t>(response.status));
+        return;
+    }
+
+    JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
+
+    jmethodID handleQueryImageMethod;
+    err = JniReferences::GetInstance().FindMethod(env, mOtaProviderDelegate, "handleQueryImage",
+                                                  "(IIJLjava/lang/Integer;Ljava/lang/String;Ljava/lang/Boolean;[B)Lchip/"
+                                                  "devicecontroller/OTAProviderDelegate$QueryImageResponse;",
+                                                  &handleQueryImageMethod);
+    VerifyOrReturn(err == CHIP_NO_ERROR, ChipLogError(Controller, "Could not find handleQueryImage method"));
+
+    jobject boxedHardwareVersion = nullptr;
+    if (hardwareVersion.HasValue())
+    {
+        chip::JniReferences::GetInstance().CreateBoxedObject<jint>(
+            "java/lang/Integer", "(I)V", static_cast<jint>(hardwareVersion.Value()), boxedHardwareVersion);
+    }
+    jobject boxedLocation = nullptr;
+    if (location.HasValue())
+    {
+        boxedLocation = env->NewStringUTF(std::string(location.Value().data(), location.Value().size()).c_str());
```

Comment:
boxedLocation is leaking localRef, please use 

Pattern:
JniLocalReferenceManager manager(env); ==> manage the lifetime for all local refs in current function.
....
jstring yy = env->NewStringUTF(xx)

example code is here, https://github.com/project-chip/connectedhomeip/blob/master/src/controller/java/AndroidCallbacks.cpp#L93C5-L93C43

Reply:
I added.